src/base/hide_password.cpp

```diff
@@ -0,0 +1,861 @@
+/*
+ *
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+
+/*
+ * hide_password.cpp -
+ */
+
+#ident "$Id$"
+
+#include <stdio.h>
+#include <stdarg.h>
+#include <stdlib.h>
+#include <time.h>
+#include <string.h>
+#include <errno.h>
+#if defined(WINDOWS)
+#include <sys/timeb.h>
+#include <process.h>
+#include <io.h>
+#else
+#include <unistd.h>
+#include <sys/time.h>
+#include <sys/timeb.h>
+#include <signal.h>
+#endif
+#include <assert.h>
+
+#include "porting.h"
+#include "cas_common.h"
+#include "hide_password.h"
+//#include "parse_tree.h"
+#include "system_parameter.h"
+//#include "environment_variable.h"
+//#include "broker_config.h"
+//#include "util_func.h"
+
+//extern PARSER_CONTEXT *this_parser;
+//==========================================================================
+
+static unsigned char is_space[0x100] =
+{
+  // ' '  '\t'  '\r'  '\n'  '\f'  '\v'
+  // 0x20 0x09  0x0d  0x0a  0x0c  0x0b
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
+};
+
+#define SKIP_SPACE_CHARACTERS(p)                \
+  do {                                          \
+        while (is_space[(unsigned char) *p])     \
+	    {                                   \
+	      p++;                              \
+	    }                                   \
+   } while(0)
+
```

Comment:
duplicated code.
```
// chartype.c
int char_isspace (int c)
{
  return ((c) == ' ' || (c) == '\t' || (c) == '\r' || (c) == '\n' || (c) == '\f' || (c) == '\v');
}

```

Reply:
We plan to maintain the status quo for the time being.
I don't want to keep calling the function to skip consecutive spaces.
Additionally, a single function call requires multiple comparisons, which I want to avoid as much as possible.

I think it's necessary to improve functions in the chartype.c,  if possible.

Reply:
First, I created the issue below.
[CBRD-25043]